lms/djangoapps/edxnotes/tests.py

```diff
@@ -144,10 +145,11 @@ def test_edxnotes_enabled(self, mock_generate_uid, mock_get_id_token, mock_get_t
                 "eventStringLimit": settings.TRACK_MAX_EVENT / 6,
             },
         }
-        self.assertEqual(
-            problem.get_html(),
-            render_to_string("edxnotes_wrapper.html", expected_context),
-        )
+        with impersonate(user):
+            self.assertEqual(
+                problem.get_html(),
+                render_to_string("edxnotes_wrapper.html", expected_context),
+            )
```

Comment:
If you had to modify this test to work with the new code, I'm concerned that it represents some compatibility break with what's currently on prod.

Reply:
The edxnotes tests are not hitting any endpoints; they're just using the runtime directly. So I had to impersonate as the user.

I'm assuming that the edxnotes decorator will be used in the context of a request. If there's a non-django-request usecase, then this refactoring won't work.

Reply:
I don't really remember how notes auth worked, because most of it lives in a separate service.